python/lsst/ts/watcher/alarm.py

```diff
@@ -0,0 +1,205 @@
+        self.name = name
+        self.callback = callback
+
+        self.severity = AlarmSeverity.NONE
+        self.max_severity = AlarmSeverity.NONE
+        self.reason = ""
+        self.acknowledged = False
+        self.acknowledged_by = ""
+        self.enabled = True
+
+        self.timestamp_severity_oldest = 0
+        self.timestamp_severity_newest = 0
+        self.timestamp_max_severity = 0
+        self.timestamp_acknowledged = 0
+
+    def __eq__(self, other):
+        for field in ("name",
+                      "callback",
+                      "severity",
+                      "max_severity",
+                      "reason",
+                      "acknowledged",
+                      "acknowledged_by",
+                      "enabled",
+                      "timestamp_severity_oldest",
+                      "timestamp_severity_newest",
+                      "timestamp_max_severity",
+                      "timestamp_acknowledged"):
+            if getattr(self, field) != getattr(other, field):
+                return False
+        return True
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def set_severity(self, severity, reason):
+        """Set the severity.
+
+        Parameters
+        ----------
+        severity : `AlarmSeverity` or `int`
+            New severity.
+        reason : `str`
+            The reason for this state; this should be a brief message
+            explaining what is wrong. Ignored if severity is NONE.
+
+        Returns
+        -------
+        updated : `bool`
+            True if the alarm state changed (i.e. if any fields were modified),
+            False otherwise.
+        """
+        severity = AlarmSeverity(severity)
+        if severity == AlarmSeverity.NONE and self.quiescent:
+            # ignore repeated NONE when alarm is already reset
+            return False
+
+        curr_tai = salobj.tai_from_utc(time.time())
+        if self.severity != severity:
+            self.timestamp_severity_oldest = curr_tai
+            self.severity = severity
+        if self.severity != AlarmSeverity.NONE:
+            self.reason = reason
+        self.timestamp_severity_newest = curr_tai
+        if self.severity == AlarmSeverity.NONE and self.acknowledged:
+            # reset the alarm
+            self.reason = ""
+            self.acknowledged = False
+            self.max_severity = AlarmSeverity.NONE
+            self.timestamp_acknowledged = curr_tai
+            self.timestamp_max_severity = curr_tai
+        elif self.severity > self.max_severity:
+            if self.acknowledged:
+                self.acknowledged = False
+                self.acknowledged_by = ""
+                self.timestamp_acknowledged = curr_tai
+            self.max_severity = self.severity
+            self.timestamp_max_severity = curr_tai
+
+        self._run_callback()
+        return True
+
+    def acknowledge(self, severity, user):
```

Comment:
in Requirement LTS-897-REQ-0001 - the watcher should publish to SAL when an alarm is acknowledged - I don't see that here- maybe elsewhere?

Reply:
The Watcher will publish the alarm event whenever the alarm state changes. When an alarm is acknowledged the alarm's `acknowledged` field is set True and LOVE will see this when the Alarm event is published.

My intent is to have the Watcher model or CSC publish events. I haven't written these yet.